playlist/playlist.go

```diff
@@ -205,6 +205,36 @@ func (p *Playlist) FindNodeInsideAdBreak(node *internal.Node) (*internal.Node, b
 	return nil, false
 }
 
+// Returns the last DateRange (#EXT-X-DATERANGE) node with SCTE35-OUT marking (i.e. the last Ad Break) in the playlist
+func (p *Playlist) FindLastAdBreak() (*internal.Node, bool) {
+	adBreaks := p.Breaks()
+	if len(adBreaks) == 0 {
+		return nil, false
+	}
+	return adBreaks[len(adBreaks)-1], true
+}
+
+// DuplicateAdBreak checks if two ad breaks have the same START-DATE, indicating a duplicate.
+// Two ad breaks are considered duplicates if they share the same START-DATE
+// and the same PLANNED-DURATION.
+func (p *Playlist) HasDuplicateAdBreak() bool {
```

Comment:
Should we check if the `DateRangeNode` has a duplication instead of checking if the playlist has any duplicated breaks?